### ISSUE TITLE
Passing a value from a modules php-File to the React component

### ISSUE THREAD

**Issue by ousiadigital** [elegantthemes/create-divi-extension#339]
### Problem Description
I'm trying to use functions which derive from a different plugin and return a string in the php-File of a Divi Builder Module. It works fine for the render Function in the php-File, but when it comes to the React component I would like to just pass the string to the component to use it to render the same output.
Is this possible with the current API?

**Comment by ArcaneDiver** [comment]
You have to make an ajax request, to retrieve data.
[AJAX WP Documentation](https://codex.wordpress.org/AJAX_in_Plugins) 

client-side
```
async fetch_posts() {

    window.jQuery.$.ajax({
      type: "GET",
      url: window.origin + "/wp-admin/admin-ajax.php",
      dataType: "json",
      data: {
        action: "get_buongiorni",
        number_of_posts: this.props.number_of_posts
      },
      success: function (response) {
        console.log("res", response);
      }
    });
  }
```


server-side

```
                add_action('wp_ajax_get_buongiorni', 'handle_request');
		add_action('wp_ajax_nopriv_get_buongiorni', 'handle_request');

		function handle_request() {

			$number = $_GET["number_of_posts"];

			if(isset($number)) {
				echo json_encode(BHUB_BuongiorniCustomModules::get_buongiorni($number));
			} else {
				echo "error";
			}

			exit;
		}
```

**Comment by dtcblyth** [comment]
There are two ways I know of doing this.

1) If your string value is static, meaning it doesn't change or depend upon any user settings, you can put your string into a hidden field, something like this:

'my_hidden_field' =>  array(
  'label' =>  esc_html__('My Invisible Label', 'my-domain'),
  'description'  =>  esc_html__('My Invisible Description', 'my-domain'),
  'option_category'  => 'configuration',
  'tab_slug'  => 'general',
  'toggle_slug'  => 'content',
  'type'  => 'hidden',
  'default' =>  my_get_string_function_here(),
),

2) If your string value is dynamic, meaning it changes or depends upon user settings, you'll need to look at using a computed field. This is Divi’s implementation of Ajax. It requires 3 interelated parts to get it working. 1) The regular user fields that 'affect' the computed property. 2) The computed field which 'depends' upon the regular fields. These begin with a double uderscore (eg.  __my_computed_field). 3) A callback function which is 'called' by the computed field and returns the value which you want to the computed field. You then get the value of the computed filed in the React component the same way you would any other value (eg. this.props.__my_computed_field; ). Have a look at some of Divi's own modules and see if you can get your head around it.

**Comment by lanresmith** [comment]
Thanks @dtcblyth for your great contribution. The two ways you mentioned should do the job.
